logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationInteractionAvailabilityUseCaseTest.kt

```diff
@@ -0,0 +1,198 @@
+
+            verify(arrangement.conversationRepository)
+                .suspendFunction(arrangement.conversationRepository::observeConversationDetailsById)
+                .with(eq(conversationId))
+                .wasInvoked(exactly = once)
+
+            awaitComplete()
+        }
+
+    }
+
+    @Test
+    fun givenAConversationId_whenGroupDetailsReturnsError_thenInteractionShouldReturnFailure() = runTest {
+        val conversationId = TestConversation.ID
+        val selfUser = TestUser.SELF
+
+        val (arrangement, observeConversationInteractionAvailability) = Arrangement()
+            .withGroupConversationError()
+            .arrange()
+
+        observeConversationInteractionAvailability(conversationId).test {
+            val interactionResult = awaitItem()
+            assertIs<IsInteractionAvailableResult.Failure>(interactionResult)
+
+            verify(arrangement.conversationRepository)
+                .suspendFunction(arrangement.conversationRepository::observeConversationDetailsById)
+                .with(eq(conversationId))
+                .wasInvoked(exactly = once)
+
+            awaitComplete()
+        }
+
+    }
+
+    @Test
+    fun givenAConversationId_whenOtherUserIsBlocked_thenInteractionShouldBeDisabled() = runTest {
+        val conversationId = TestConversation.ID
+        val selfUser = TestUser.SELF
+
+        val (arrangement, observeConversationInteractionAvailability) = Arrangement()
+            .withBlockedUserConversation()
+            .arrange()
+
+        observeConversationInteractionAvailability(conversationId).test {
+            val interactionResult = awaitItem()
+            assertEquals(IsInteractionAvailableResult.Success(InteractionAvailability.BLOCKED_USER), interactionResult)
+
+            verify(arrangement.conversationRepository)
+                .suspendFunction(arrangement.conversationRepository::observeConversationDetailsById)
+                .with(eq(conversationId))
+                .wasInvoked(exactly = once)
+
+            awaitComplete()
+        }
+
+    }
+
+    @Test
+    fun givenAConversationId_whenOtherUserIsDeleted_thenInteractionShouldBeDisabled() = runTest {
+        val conversationId = TestConversation.ID
+        val selfUser = TestUser.SELF
+
+        val (arrangement, observeConversationInteractionAvailability) = Arrangement()
+            .withDeletedUserConversation()
+            .arrange()
+
+        observeConversationInteractionAvailability(conversationId).test {
+            val interactionResult = awaitItem()
+            assertEquals(IsInteractionAvailableResult.Success(InteractionAvailability.DELETED_USER), interactionResult)
+
+            verify(arrangement.conversationRepository)
+                .suspendFunction(arrangement.conversationRepository::observeConversationDetailsById)
+                .with(eq(conversationId))
+                .wasInvoked(exactly = once)
+
+            awaitComplete()
+        }
+
+    }
```

Comment:
For the future, keep in mind that the given/when in the test names are inverted, for example:

- Given: Other user is deleted
- When: Getting interaction for conversation / Invoking
- Then: Interaction should be disabled

Not only it reflects the test:

- Set up mocks or fakes
- Perform the action (call `observeConversationInteractionAvailability(conversationId)`)
- Assert results


But it is also easier to find what we want. When scrolling through the list of tests, the beginning of the test name can give more information about what the scenario that's being tested. Instead of just the same `givenAConversationId`.

Reply:
Good point, changed 👍